File: src/test/resources/org/springframework/scripting/js/UserService.js
importPackage(java.util);
importPackage(org.springframework.scripting.js);

var log = org.apache.commons.logging.LogFactory.getLog("org.springframework.scripting.js.JavaScriptUserService");

function validate(users) {
	log.debug("Validating: " + users);
	var entries = users.entrySet().toArray();
	var ids = [];
	for(entry in entries) {
		ids.push(entries[entry].key);
	}
	return new HashSet(Arrays.asList(ids));
}

function loadByIds(ids) {
	var result = new ArrayList();
	for(id in ids) {
		result.add(new User().withName(id));
	}
	return result;
}

function dump(user) {
	return "User[name=" + user.getName() + ", country=" + dumpCountry(user.getCountry()) + "]";
}

function dumpCountry(country) {
	return "Country[capital=" + country.getCapital() + "]";
}

File: src/test/java/org/springframework/scripting/js/UserService.java
package org.springframework.scripting.js;

import java.util.List;
import java.util.Map;
import java.util.Set;

/**
 * @author Tomasz Nurkiewicz
 * @since 2010-09-21, 19:35:37
 */
public interface UserService {

	Set<Integer> validate(Map<Integer, User> users);

	List<User> loadByIds(int[] ids);

	String dump(User user);

}
